kubernetes/schema_endpointslice.go

```diff
@@ -0,0 +1,152 @@
+// SPDX-License-Identifier: MPL-2.0
+
+package kubernetes
+
+import (
+	"errors"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"k8s.io/apimachinery/pkg/util/validation"
+)
+
+func schemaEndpointSliceSubsetEndpoints() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"addresses": {
+				Type:        schema.TypeList,
+				Description: "addresses of this endpoint. The contents of this field are interpreted according to the corresponding EndpointSlice addressType field.",
+				Required:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"condition": {
+				Type:        schema.TypeList,
+				Description: "condition contains information about the current status of the endpoint.",
+				Required:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"ready": {
+							Type:        schema.TypeBool,
+							Description: "ready indicates that this endpoint is prepared to receive traffic, according to whatever system is managing the endpoint.",
+							Optional:    true,
+						},
+						"serving": {
+							Type:        schema.TypeBool,
+							Description: "serving is identical to ready except that it is set regardless of the terminating state of endpoints.",
+							Optional:    true,
+						},
+						"terminating": {
+							Type:        schema.TypeBool,
+							Description: "terminating indicates that this endpoint is terminating.",
+							Optional:    true,
+						},
+					},
+				},
+			},
+			"hostname": {
+				Type:        schema.TypeString,
+				Description: "hostname of this endpoint. This field may be used by consumers of endpoints to distinguish endpoints from each other.",
+				Optional:    true,
+				ValidateFunc: func(v interface{}, k string) ([]string, []error) {
+					hostname := v.(string)
+					errs := []error{}
+					errLabels := validation.IsDNS1123Label(hostname)
+					for _, e := range errLabels {
+						errs = append(errs, errors.New(e))
+					}
+
+					return nil, errs
+				},
+			},
+			"node_name": {
+				Type:        schema.TypeString,
+				Description: "nodeName represents the name of the Node hosting this endpoint. This can be used to determine endpoints local to a Node.",
+				Optional:    true,
+			},
+			"target_ref": {
+				Type:        schema.TypeList,
+				Description: "targetRef is a reference to a Kubernetes object that represents this endpoint.",
+				MaxItems:    1,
+				Optional:    true,
+				Elem:        schemaObjectReference(),
+			},
+			"zone": {
+				Type:        schema.TypeString,
+				Description: "zone is the name of the Zone this endpoint exists in.",
+				Optional:    true,
+			},
+		},
+	}
+}
+
+func schemaEndpointSliceSubsetPorts() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"port": {
+				Type:        schema.TypeString,
+				Description: "port represents the port number of the endpoint.",
+				Required:    true,
```

Comment:
You'll want to validate the value is actually a positive integer, like this: https://github.com/hashicorp/terraform-provider-kubernetes/blob/5bc27bc9b9e2831cb43d838bf9017f14130dca65/kubernetes/schema_job_spec.go#L159-L165